src/formatDate.js

```diff
@@ -8,7 +8,37 @@
  * @returns {string}
  */
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const splittedDate = date.split(fromFormat[3]);
+
+  const yearFormat = fromFormat.includes('YY') ? 'YY' : 'YYYY';
+
+  const year = splittedDate[fromFormat.indexOf(yearFormat)];
+  const month = splittedDate[fromFormat.indexOf('MM')];
+  const day = splittedDate[fromFormat.indexOf('DD')];
+
+  const newYearFormat = toFormat.includes('YY') ? 'YY' : 'YYYY';
+
+  const newIndexOfYear = toFormat.indexOf(newYearFormat);
+  const newIndexOfMonth = toFormat.indexOf('MM');
+  const newIndexOfDay = toFormat.indexOf('DD');
+
+  const newDate = [];
+
+  let formatedYear = year;
+
+  if (yearFormat.length - newYearFormat.length === 2) {
+    formatedYear = year.slice(2, 4);
+  }
+
+  if (yearFormat.length - newYearFormat.length === -2) {
+    formatedYear = +year >= 30 ? '19' + year : '20' + year;
+  }
```

Comment:
The conversion from 'YY' to 'YYYY' is correct, but it would be more clear and maintainable to explicitly check the formats ('YY' to 'YYYY' or vice versa) instead of relying on the length difference.